code/modules/mob/living/simple_animal/abnormality/_abnormality.dm

```diff
@@ -177,6 +177,9 @@
 	if(secret_chance && (prob(1)))
 		InitializeSecretIcon()
 
+	if(SSmaptype.chosen_trait == FACILITY_TRAIT_JOKE_ABNOS && secret_chance)
+		InitializeSecretIcon()
```

Comment:
Tbh i think this would be cleaner
```suggestion
	if(secret_chance && (prob(1) || SSmaptype.chosen_trait == FACILITY_TRAIT_JOKE_ABNOS))
		InitializeSecretIcon()
```